verifiers/utils/builder_test.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	serrors "github.com/slsa-framework/slsa-verifier/errors"
+)
+
+func Test_ParseBuilderID(t *testing.T) {
```

Comment:
could you put in 1-2 tests on matches? just for match with/without version

Reply:
yes, those are in there now, including for `BuilderIDNew()`. PTAL